src/lists.c

```diff
@@ -1760,20 +1756,8 @@ void            AssListLevel (
     Obj pos,pos1,pos2;
 
     /* check <objs>                                                        */
-    while ( ! IS_DENSE_LIST(objs) || LEN_LIST(lists) != LEN_LIST(objs) ) {
-        if ( ! IS_DENSE_LIST(objs) ) {
-            objs = ErrorReturnObj(
-                "List Assignment: <objs> must be a dense list (not a %s)",
-                (Int)TNAM_OBJ(objs), 0L,
-                "you can replace <objs> via 'return <objs>;'" );
-        }
-        if ( LEN_LIST(lists) != LEN_LIST(objs) ) {
-            objs = ErrorReturnObj(
-         "List Assignment: <objs> must have the same length as <lists> (%d)",
-                LEN_LIST(lists), 0L,
-                "you can replace <objs> via 'return <objs>;'" );
-        }
-    }
+    CheckIsDenseList("List Assignment", "objs", objs);
+    CheckSameLength("List Assignment", "objs", "lists", objs, lists);
```

Comment:
Just to be explicit about it: Here we loose some flexibility, in that previously we allowed the user to `return <obj>` a new list of objects, and now we error out. I thought about restoring this here (and in a few other similar places), but it just didn't seem worth the effort. If somebody disagrees, let me know.